File: Shortest Uncommon Subsequence - GFG/shortest-uncommon-subsequence.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String A[] = read.readLine().split(" ");
            
            String S = A[0];
            String T = A[1];

            Solution ob = new Solution();
            System.out.println(ob.shortestUnSub(S,T));
        }
    }
}// } Driver Code Ends


//bcda
//bcd


//dp[i][j]=Math.min(dp[i-1][j],dp[i-1][k]+ 1)


//User function Template for Java

class Solution {
    static int shortestUnSub(String S, String T) {
        // code here
        int n=S.length();
        int m=T.length();
        int[][]dp=new int[n+1][m+1];
        
        for(int i=0;i<=n;i++){
            dp[i][0]=1;
        }
        
        int MAX=501;
        for(int j=0;j<=m;j++){
            dp[0][j]=MAX;
        }
        
        
        for(int i=1;i<=n;i++){
            for(int j=1;j<=m;j++){
                int k;
                for(k=j-1;k>=0;k--){
                    if(S.charAt(i-1)==T.charAt(k))break;
                }
                
                if(k==-1){
                    dp[i][j]=1;
                }else{
                    dp[i][j]=Math.min(dp[i-1][j],dp[i-1][k]+ 1);
                }
            }
        }
        
        if(dp[n][m]>=MAX)return -1;
        
        return dp[n][m];
        
    }
};

File: Shortest Uncommon Subsequence - GFG/shortest-uncommon-subsequence.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
class Solution {
  public:
      int findShortest(vector<vector<int>> &dp, string &S, string &T, int n, int m, int i, int j){
        if(i >= n){
            return 501;
        }
        if(j >= m){
            return 1;
        }
        if(dp[i][j] != -1){
            return dp[i][j];
        }
        int k = j;
        for(; k < m; k++){
            if(S[i] == T[k]){
                break;
            }
        }
        if(k == m){
            return 1;
        }
        return dp[i][j] = min(findShortest(dp, S, T, n, m, i + 1, j), 1 + findShortest(dp, S, T, n, m, i + 1, k + 1));
    }
  
  
    int shortestUnSub(string S, string T) {
        int n = S.size();
        int m = T.size();
        vector<vector<int>> dp (n + 1, vector<int> (m + 1, -1));
        int ans = findShortest(dp, S, T, n, m, 0, 0);
        return ans >= 501 ? -1 : ans;
    }
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        string S,T;
        cin>>S>>T;

        Solution ob;
        cout << ob.shortestUnSub(S,T) << endl;
    }
    return 0;
}  // } Driver Code Ends